File: sliding-window/best-time-stocks.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n = len(prices)
        left = 0
        right = 1
        maxProfit = 0

        while right < n:
            if prices[left] < prices[right]:
                maxProfit = max(maxProfit, prices[right] - prices[left])
            else:
                left = right
            right += 1

        return maxProfit

File: sliding-window/best-time-stocks.cpp
#include <vector>

using namespace std;

class Solution
{
public:
    int maxProfit(vector<int> &prices)
    {
        int n = prices.size();
        int buyP = prices[0];
        int maxP = 0;

        for (int right = 0; right < n; right++)
        {
            maxP = max(maxP, prices[right] - buyP);
            buyP = min(buyP, prices[right]);
        }

        return maxP;
    }
};
